test/hpke_test.c

```diff
@@ -0,0 +1,1432 @@
+    if (!TEST_true(OSSL_HPKE_encap(ctx, enc, &enclen, pub, publen, NULL, 0)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_seal(ctx, cipher, &cipherlen, NULL, 0,
+                                  plain, plainlen)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_export(ctx, exp, 32,
+                                    (unsigned char *)estr, strlen(estr))))
+        goto end;
+    /* check a 2nd call with same input gives same output */
+    if (!TEST_true(OSSL_HPKE_export(ctx, exp2, 32,
+                                    (unsigned char *)estr, strlen(estr))))
+        goto end;
+    if (!TEST_true(TEST_mem_eq(exp, 32, exp2, 32)))
+        goto end;
+    if (!TEST_ptr(rctx = OSSL_HPKE_CTX_new(hpke_mode, hpke_suite,
+                                           testctx, NULL)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_decap(rctx, enc, enclen, privp, NULL, 0)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_open(rctx, clear, &clearlen, NULL, 0,
+                                  cipher, cipherlen)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_export(rctx, rexp, 32,
+                                    (unsigned char *)estr, strlen(estr))))
+        goto end;
+    /* check a 2nd call with same input gives same output */
+    if (!TEST_true(OSSL_HPKE_export(rctx, rexp2, 32,
+                                    (unsigned char *)estr, strlen(estr))))
+        goto end;
+    if (!TEST_true(TEST_mem_eq(rexp, 32, rexp2, 32)))
+        goto end;
+    if (!TEST_true(TEST_mem_eq(exp, 32, rexp, 32)))
+        goto end;
+    OSSL_HPKE_CTX_free(ctx);
+    OSSL_HPKE_CTX_free(rctx);
+    EVP_PKEY_free(privp);
+    return 1;
+end:
+    OSSL_HPKE_CTX_free(ctx);
+    OSSL_HPKE_CTX_free(rctx);
+    EVP_PKEY_free(privp);
+    return 0;
+}
+
+/**
+ * @brief Check mapping from strings to HPKE suites
+ * @return 1 for success, other otherwise
+ */
+static int test_hpke_suite_strs(void)
+{
+    int overallresult = 1;
+    int kemind = 0;
+    int kdfind = 0;
+    int aeadind = 0;
+    int sind = 0;
+    char sstr[128];
+    OSSL_HPKE_SUITE stirred;
+
+    for (kemind = 0;
+         kemind != (sizeof(kem_str_list) / sizeof(char *));
+         kemind++) {
+        for (kdfind = 0;
+             kdfind != (sizeof(kdf_str_list) / sizeof(char *));
+             kdfind++) {
+            for (aeadind = 0;
+                 aeadind != (sizeof(aead_str_list) / sizeof(char *));
+                 aeadind++) {
+                snprintf(sstr, 128, "%s,%s,%s",
+                         kem_str_list[kemind],
+                         kdf_str_list[kdfind],
+                         aead_str_list[aeadind]);
+                if (TEST_true(OSSL_HPKE_str2suite(sstr, &stirred)) != 1) {
+                    overallresult = 0;
+                }
+            }
+        }
+    }
+    for (sind = 0;
+         sind != (sizeof(bogus_suite_strs) / sizeof(char *));
+         sind++) {
+        char dstr[128];
+
+        sprintf(dstr, "str2suite: %s", bogus_suite_strs[sind]);
```

Comment:
What is the point of this sprintf?

Reply:
nothing! (any more:-) it was previously used for tracing